ui/components/data-sources/BaseDataSourceDialog.js

```diff
@@ -151,9 +151,33 @@ class BaseDataSourceDialog extends Component {
         throw new Error(`isDisabled(controlId: string) not implemented in ${this.constructor.name}`);
     }
 
+    /**
+     * Test the data source and show message on the dialog.
+     */
+    onTest() {
+        this.testDataSource()
+            .then(() => this.setState({ success: "Connection to data source is successful" }))
+            .catch((message) => this.setState({ err: message }));
+    }
+
+    testDataSource() {
+        const { name, type } = this.state;
+
+        const config = { options: this.getConfigByType(type) };
+        // TODO: what here?
+        // TODO: how to call `getDataSourceTestResult` query?
+
```

Comment:
I don't know how to call the `getDataSourceTestResult` query within this component.

@josemigallas any help?

Normally, data source dialog subclass instances are bound to a single mutation. But, in this case, this is an additional functionality which has nothing to do with the mutation.

I tried this:
```
import gql from "graphql-tag";

client
  .query({
    query: gql`
      {
        rates(currency: "USD") {
          currency
        }
      }
    `
  })
  .then(result => console.log(result));
```
which is simply using the ApolloClient with nothing specific to React, but I don't know how to get a reference to `client`. 
@josemigallas does this look good to you? or, is there any other approach you would recommend?

Reply:
@aliok One possibility is to use a `<Query>` tag. See here for an example: https://github.com/aerogear/data-sync-ui/blob/master/ui/components/resolvers/ResolversList.js#L94

Reply:
wow that looks ugly :smiling_imp:  :laughing: 
will give it a try

Reply:
Another possibility is to import `graphql` from `react-apollo` and use this. I'll give it a try and let you know.